### PR TITLE
fix(cockpit): silent-orphan watchdog suppression for background Bash + ScheduleWakeup

### LINKED ISSUE #1401: Cockpit: silent-orphan watchdog kills turns waiting on backgrounded Bash or pending ScheduleWakeup
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Problem

The silent-orphan watchdog still kills turns where the agent is legitimately waiting on Claude SDK background features that aoe doesn't model: `Bash(run_in_background: true)` and `ScheduleWakeup`. #1360 added a `is_async` floor for the Claude SDK `Agent` tool, but the same root-cause shape recurs for these other two waiting patterns, which are far more common (`/loop` dynamic sessions, "let cargo build run while I think" flows, parallel local tasks).

Session `65c7bd0f22424242` shows 4 `prompt_orphaned` events (seq 1791, 3033, 4998, 5146). Each one followed the same shape: agent fires backgrounded Bash and / or ScheduleWakeup, finishes its status message, then the daemon kills the turn 20-120s later despite the work being intentional and tracked.

## Evidence from the affected session

The most recent `prompt_orphaned` (seq 5146):

```
5121 ToolCallStarted    Terminal (Bash)
5122 ToolCallUpdated    "npm install 2>&1 | tail -5"  run_in_background=true
5125 RawAgentUpdate     _meta.claudeCode.toolResponse.backgroundTaskId="biqa58ekr"
5126 ToolCallCompleted  "Command running in background with ID: biqa58ekr..."
5128 ToolCallStarted    ScheduleWakeup
5129 ToolCallUpdated    delaySeconds=240  reason="check cargo build + npm install progress"
5130 WakeupScheduled    at="2026-05-22T07:31:55.963Z"  reason="check cargo build + npm install progress"
5132 RawAgentUpdate     _meta.claudeCode.toolResponse.scheduledFor=1779435120000
5133 ToolCallCompleted  "Next wakeup scheduled for 09:32:00 (in 244s)."
5135 AgentMessageChunk  ""
5136 AgentMessageChunk  "Waiting for c"
5137 AgentMessageChunk  "argo build + npm install. ETA ~4min. Will resume when ready."
5138 UsageUpdated       cost: null
5139 UsageUpdated       cost: { amount: 13.85, currency: "USD" }
5140 UsageUpdated       cost: null
5141 ThinkingStarted
5142 AgentMessageChunk  ""
5143 AgentMessageChunk  "npm done. Wait for cargo."
5144 UsageUpdated       cost: null
5145 UsageUpdated       cost: { amount: 14.21, currency: "USD" }
5146 Stopped            reason: "prompt_orphaned"
5147 Stopped            reason: "user_stopped"
```

Earlier in the same turn there were two more backgrounded Bash launches still alive at the orphan moment:

```
5077 "date && cargo build --features serve 2>&1 | tail -5 && date"
5080 backgroundTaskId="bzzgulffz"
5090 "until [ -f target/debug/aoe ] && ! pgrep -f \"cargo build\" ..."
5093 backgroundTaskId="bgxe33hwb"
```

Timing (ms since epoch on the event rows):

- last `Progress` event: 5143 at 1779434895569 (07:28:15.569 UTC).
- cost-populated `UsageUpdated`: 5145 at 1779434896184 (07:28:16.184).
- `prompt_orphaned`: 5146 at 1779434926501 (07:28:46.501).

Delta from last progress to orphan: ~30.9s. The cost-populated update at 5145 triggers `LifecycleSignal::TerminalUsage`, switching the effective grace from the 120s base to the 20s fast grace; the watchdog polls every 5s, so firing at last_progress + 20s + (one polling tick) is the expected behaviour. The bug is upstream of the timer: the daemon should never have armed this watchdog in the first place, because the agent had three live background tasks and a wakeup scheduled three minutes in the future.

Across the session: 4 `prompt_orphaned` out of ~30 turns. Every one of them is a false-positive on the same shape (backgrounded Bash and / or pending ScheduleWakeup, no live ACP-visible `ToolCall`).

## Root cause

`classify_lifecycle_signal` at `src/cockpit/acp_client.rs:329-365` decides what counts as in-flight tool work for the watchdog. It looks at `SessionUpdate::ToolCall` / `ToolCallUpdate` lifecycle only; it has no awareness of two Claude SDK features the daemon already sees:

1. **Backgrounded Bash.** When the agent calls `Bash(run_in_background: true)`, claude-agent-acp emits `ToolCallStatus::Completed` immediately (the foreground call returns instantly with a backgroundTaskId), and the actual subprocess keeps running off-protocol. The completion content includes the marker text `"Command running in background with ID: <id>"`, and the raw `_meta.claudeCode.toolResponse.backgroundTaskId` carries the id. The daemon strips this back to a generic `ToolCallCompleted` and the watchdog sees zero in-flight tool work.

2. **Pending ScheduleWakeup.** `wakeup_event_from_raw` at `src/cockpit/acp_client.rs:1812-1859` already parses `delaySeconds` from the tool args and emits `Event::WakeupScheduled { at, reason }` to the cockpit event store (so #1091's "Asleep until..." banner works). But that event never reaches the `LifecycleSignal` mpsc that drives the watchdog: the watchdog state machine at `acp_client.rs:329-365` has no `WakeupPending` variant, and the prompt loop has no `next_wakeup_at` field.

So the daemon already knows (a) which Bash tool calls were backgrounded and (b) when the next wakeup will fire. It just throws both signals away before the watchdog can use them.

The #1360 `async_agent_running` floor only covers the `Agent` tool with `isAsync: true` (detected at `acp_client.rs:382-389` via the `"Async agent launched successfully"` content prefix). That's a narrower feature than backgrounded Bash and orthogonal to ScheduleWakeup; both predate `is_async` and are used heavily by users (`/loop` dynamic, long builds, multi-process work).

## Proposal

Three layers. (A) is the minimum viable fix, (B) is the structural one, (C) is the UX guardrail.

### A. Bridge backgrounded Bash and pending ScheduleWakeup into the watchdog state

Two new `LifecycleSignal` variants:

```rust
enum LifecycleSignal {
    // ...existing...
    /// A tool call completed with a `run_in_background: true` Bash launch.
    /// Holds a synthetic id so the prompt loop can keep `tool_calls_in_flight`
    /// non-empty until the agent either polls the task (BashOutput, KillShell)
    /// or the turn ends.
    BackgroundTaskStarted(String),
    /// `Event::WakeupScheduled` was emitted from a `ScheduleWakeup` tool.
    /// Suppresses the watchdog until `at + small_grace`, after which the
    /// watchdog rearms with the normal base grace (the wakeup fired or
    /// failed, and the agent should produce progress soon either way).
    WakeupPending { at: DateTime<Utc> },
}
```

Detection sites:

- `classify_lifecycle_signal` (`acp_client.rs:329-365`): on `ToolCallUpdate { status: Completed }`, inspect `update.fields.content` for the `"Command running in background with ID: "` prefix and emit `BackgroundTaskStarted(<task_id>)` instead of (or in addition to) `ToolCompleted`.
- The same site or the `ScheduleWakeup` raw_input parsing at `acp_client.rs:2121-2130`: when the `WakeupScheduled` event is built, also push `WakeupPending { at }` onto the `LifecycleSignal` channel.

Prompt-loop changes (around the silent-orphan select arm, `acp_client.rs:2984-3047` per #1360's coordinates):

- Add a `background_tasks: HashSet<String>` and a `next_wakeup_at: Option<Instant>` to the prompt-loop state.
- When evaluating "should watchdog fire?", treat both fields as suppression: `tool_calls_in_flight.is_empty() && background_tasks.is_empty() && next_wakeup_at.map_or(true, |t| t < now())`.
- Clear `background_tasks` entries when the agent calls `BashOutput`, `KillShell`, or emits a `ToolCallUpdate` that references the background id (the SDK includes the id in the input args of those follow-up tools).
- Clear `next_wakeup_at` when a new `UserPromptSent` arrives (wakeup fired and turned into a new prompt) or when it expires (no longer suppresses; watchdog rearms).

### B. Generalize "off-protocol async work" into a single signal

(A) hardcodes detection for two specific Claude SDK features. The architectural shape is "claude-agent-acp completes the visible ToolCall but the work continues off-protocol." That set will keep growing: future SDK features (CronCreate auto-fire, batch tools, sub-shells) will need the same treatment.

Cleaner shape: extend the `LifecycleSignal` classifier with a single `OffProtocolWorkStarted { kind, id }` variant fed by a small handler-side helper `detect_off_protocol_work(content, raw_meta)` that knows the current set of markers. Async-agent (`is_async: true`), backgrounded Bash, and ScheduleWakeup all funnel through the same path; future detectors are one helper change away. This is roughly the shape #1360 already started for `is_async`; expanding it covers the other two without a new mechanism.

The cleanest long-term home is upstream: claude-agent-acp emits a structured `SessionUpdate::WorkInFlight { kind, id, expected_resume_at? }` so aoe doesn't name-sniff. Same pattern called out in #1091 (E) and the `#1041/#1043/#1059/#1062/#1064` umbrella. Out of scope for this issue, but the aoe-side bridge should be shaped so the upstream signal slots in cleanly if it lands.

### C. Loud failure mode when the watchdog does fire

Even with (A) in place, the watchdog will still occasionally false-fire on patterns we haven't detected yet. Today, `prompt_orphaned` shows up only as a brief "Reconnect" banner and the session goes to `Stopped`. The user's mental model is "I had backgrounded tasks, where did my turn go?"

When the daemon emits `Stopped { reason: "prompt_orphaned" }` and the recent event history contains a live `WakeupScheduled` future-`at` or a `backgroundTaskId` whose corresponding `BashOutput` / `KillShell` was never called, prepend a context line to the banner: `"Watchdog killed turn but 3 background tasks were still running and a wakeup was scheduled for 07:31:55. Increase cockpit.silent_orphan_grace_secs in settings, or disable it for this profile."`

Touch: `src/cockpit/supervisor.rs` (drain path, around `:1149`) and `web/src/components/cockpit/WorkerRestartingBanner.tsx` (new variant string).

## Why now

- Restart-loop is back, just with a different trigger. #1360 closed the `isAsync: true` case but `/loop` dynamic and backgrounded Bash are the more common shapes; one prod-shaped session is throwing 4 orphans across ~30 turns.
- Loss isn't just cosmetic: the agent's reattach doesn't reconnect to its background tasks. `biqa58ekr`, `bzzgulffz`, etc., keep running on the host (they're decoupled subprocesses), but the agent has no handle to poll them via `BashOutput` after the worker respawns; their stdout drains into `/tmp/claude-501/...` and the agent doesn't know which IDs to ask about.
- The bridge is local and self-contained. The classifier and prompt loop already exist; (A) is one new helper + two new variants + the suppression check.
- `/loop` dynamic mode is becoming load-bearing for long-running workflows (autonomous loops, multi-model fan-out). The bug will get more common, not less.

## Open questions

- Should `BackgroundTaskStarted` suppress the watchdog forever (until turn ends), or for a bounded window (e.g., 30 minutes like the `is_async` floor)? Forever risks pinning a turn open if a backgrounded process hangs and the agent forgets to poll. Bounded matches `is_async` precedent.
- For `WakeupPending`, the watchdog should rearm shortly after `at` even if no new progress arrives, because the wakeup either fired (new prompt) or failed (agent should produce a signal). What's "shortly"? `at + base_grace` (120s) seems safe.
- Where does the background-task suppression list live across `aoe serve` restarts? Probably nowhere: a daemon restart already resets all the in-memory watchdog state, and on respawn the worker replays the ACP session which doesn't include backgrounded subprocesses. Acceptable degradation: after a daemon restart, the watchdog may false-fire once if the agent is still mid-wait. Don't try to persist.
- Is the `"Command running in background with ID: "` prefix stable? It comes from the Claude SDK's Bash tool result formatter, not claude-agent-acp. If the prefix changes, (A) silently degrades to the current behaviour, which is survivable. Add a regression-style test under `tests/cockpit_silent_orphan.rs` so the detector breakage is caught early.

## Out of scope

- Re-attaching background task handles after a worker respawn. Separate feature; tracked alongside #1052 (mid-turn daemon restart).
- Building first-class `Status::AsleepUntil` / `Status::WaitingOnBackground` session states (covered by #1091 and adjacent UI work). This issue is about not killing those sessions, not about decorating them.
- Reworking `silent_orphan_fast_grace_secs` semantics. The accelerator is correct for the original adapter-wedge case; the bug is purely that the watchdog arms when it shouldn't.
- Surfacing `BashOutput` / `KillShell` activity in the cockpit UI. Orthogonal rendering work.

## Cross-references

- #1240 — original silent-orphan watchdog.
- #1360 — closed; added `is_async` floor for Claude SDK `Agent` tool. Same root-cause shape, different SDK feature.
- #1091 — ScheduleWakeup / Cron* UI; the `Event::WakeupScheduled` plumbing already exists, this issue just hooks it into the watchdog.
- #1216 — resume-idle watchdog cousin (same architectural problem, different timer).
- #1041 / #1043 / #1059 / #1062 / #1064 — the broader "structured work hidden in generic tool metadata" umbrella; backgrounded Bash and pending wakeups are two more instances.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

The silent-orphan watchdog (#1240, #1360) was killing turns where the agent was legitimately waiting on Claude SDK off-protocol work. Production session `65c7bd0f22424242` showed 4 `Stopped { reason: "prompt_orphaned" }` events in 30 turns. Each one fired ~30s after the agent emitted a status chunk like "Waiting for cargo build + npm install. ETA ~4min", because the daemon had no awareness that three backgrounded Bash tasks (`run_in_background: true`) and a 240s ScheduleWakeup were still in flight. The cost-populated `UsageUpdated` accelerator then collapsed the grace to 20s, sealing the false positive.

This PR extends the watchdog state machine to know about both off-protocol shapes the daemon was already receiving but ignoring:

- **Backgrounded Bash.** Detected at `ToolStarted` via `raw_input.run_in_background = true` AND at `ToolCompleted` via the content marker `"Command running in background with ID: "`. Either signal alone is enough to mark the prompt as having off-protocol work; both together are defense in depth so a single SDK string drift can't reintroduce the false-positive class.

- **Pending ScheduleWakeup.** Derived from the existing `Event::WakeupScheduled` (#1091 already parses `delaySeconds` from `raw_input`). Suppresses the watchdog until `at + base_grace`, computed as a monotonic `Instant` deadline at signal receipt so wall-clock jumps don't perturb suppression. Multiple wakeups extend (never shorten) the deadline.

Both signals lift the effective grace to `OFF_PROTOCOL_WORK_GRACE_FLOOR` (30 minutes), the same floor the async-agent fix (#1360) introduced. The watchdog still recovers from genuine adapter wedges, just slower.

The watchdog state machine was extracted into a free-standing `SilentOrphanWatchdog` struct. `apply_signal` and `should_fire` take injected `Instant` + `DateTime<Utc>` inputs, so the 13 new unit tests can drive synthetic clocks deterministically instead of relying on `tokio::time::sleep` through the integration shim. The select arm in the prompt loop is now a thin wrapper.

Two integration tests (`silent_orphan_suppressed_during_background_bash`, `silent_orphan_suppressed_during_scheduled_wakeup`) reproduce the exact production shapes via the test shim, including the cost-populated `UsageUpdate` that triggers the fast grace path; a regression that lets cost-seen shadow off-protocol suppression would false-fire well inside the 2s drain.

Lifecycle delivery is now non-lossy for every state-bearing variant. `Progress` keeps `try_send` for the streaming-chunk hot path but falls back to awaited `send` when the channel is full, so a dropped `Progress` after a `TerminalUsage` cannot leave the watchdog with stale `last_progress_at` and `cost_seen = true` (concrete false-positive scenario surfaced by the multi-model design review). Watchdog signals also deliver before the user-visible event vector so a backpressured `event_tx.send().await` can't let the polling tick fire before the suppression-bearing signal arrives.

A two-round /debate (gpt-5.5 + gemini-3.1-pro-preview) shaped the design and caught several bugs before they landed. A post-implementation review pass caught three more (failed-tool spurious suppression, `is_background_task` clobber on `InProgress` re-stamp, lifecycle/event delivery ordering); each has a dedicated regression test.

Closes #1401.

## PR Type

- [ ] New Feature
- [x] Bug Fix
- [x] Refactor
- [x] Documentation
- [ ] Infrastructure / CI

## How to test

User-runnable behavioral steps:

- [x] `cargo test --features serve --lib cockpit::acp_client` passes 66 tests including 13 new `watchdog_*` and `classify_*` cases.
- [x] `cargo test --features serve --test integration cockpit_silent_orphan` passes 6 tests (4 existing + 2 new).
- [x] In a real cockpit session, send a prompt that runs `Bash(run_in_background: true)` (e.g. ask the agent to start a long-running command in the background, then idle). The session must NOT transition to `Stopped { reason: "prompt_orphaned" }` within the configured grace window.
- [x] In a `/loop` dynamic session that calls `ScheduleWakeup` with `delaySeconds: 240`, the session must stay alive past the watchdog's base grace and the wakeup must fire normally.
- [x] Existing `prompt_orphaned` wedge recovery still works: set `AOE_COCKPIT_SIMULATE_ORPHAN_NEXT_PROMPT=1` (debug builds) before a prompt and confirm the watchdog still fires within the configured grace.
- [x] `cockpit.silent_orphan_grace_secs = 0` still fully disables the watchdog.

## Checklist

- [x] I understand the code I am submitting
- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording (no UI changes)

## AI Usage

- [ ] No AI was used
- [x] AI was used for drafting/refactoring
- [ ] This is fully AI-generated

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires /aoe-implement skill); design pass via /debate against gpt-5.5 and gemini-3.1-pro-preview; post-impl review pass against the same two.

**Any Additional AI Details you'd like to share:**

Investigation, plan, and implementation drafted by Claude under my direction. I reviewed each commit, made the design calls (extract the state machine, defense-in-depth detection at both ToolStarted and ToolCompleted, wakeup tail = at + base_grace as monotonic Instant, non-lossy delivery for state-bearing signals), and ran the manual test steps. The post-impl review caught three correctness gaps that I asked Claude to fix in a follow-up commit before opening this PR.

- [x] I am an AI Agent filling out this form (check box if true)

## Overview

This PR fixes a timing bug where the silent-orphan watchdog was prematurely terminating prompts while the agent was legitimately waiting on off-protocol work (backgrounded Bash commands and scheduled wakeups).

What changed (high-level)
- Extracted the silent-orphan watchdog into a standalone state machine with deterministic time injection for testing.
- Added detection and suppression for two off-protocol shapes:
  - Background Bash tasks (via run_in_background flag and command-completion marker).
  - Pending ScheduleWakeup calls (tracked as a monotonic deadline).
- When either off-protocol condition is present, the watchdog’s effective grace is raised to a 30-minute floor to avoid false positives.
- Made lifecycle-signal delivery non-lossy and epoch-scoped so signals won’t leak between prompts.
- Added integration and unit tests and a hermetic EnvGuard to prevent test interference.
- Updated docs to describe the new off-protocol suppression behavior.

Technical details
- New SilentOrphanWatchdog state machine takes injected Instant and DateTime<Utc> for deterministic unit tests (13 new unit tests).
- LifecycleSignal enum reworked to include structured shapes:
  - ToolStarted { id, is_background_task: bool }
  - ToolCompleted { id, succeeded, off_protocol_work: Option<OffProtocolWorkKind> }
  - LifecycleEnvelope { epoch, signal } to avoid cross-prompt contamination.
- OffProtocolWorkKind distinguishes AsyncAgent vs BackgroundCommand.
- Detection:
  - Background: recorded on ToolStarted if raw_input.run_in_background = true and/or on ToolCompleted when output contains the anchored marker "Command running in background with ID: " (marker matching tightened to start-of-line to avoid echoes).
  - Wakeup: derived from ScheduleWakeup lifecycle signals (non-failed completions) and tracked as Instant deadlines; multiple wakeups extend deadline.
- Grace semantics: presence of off-protocol work raises grace to OFF_PROTOCOL_WORK_GRACE_FLOOR (30 minutes); suppression is bounded and re-arms after expiry.
- Signal sending made robust: Progress uses try_send with a blocking send fallback to avoid losing lifecycle updates.
- Prompt loop instantiates watchdog per prompt and queries watchdog.should_fire; cancel escalation behavior preserved but driven by the state machine outcome.

Tests and docs
- 13 unit tests for watchdog behaviors and detection edge-cases.
- 2 integration tests reproducing background-Bash and ScheduleWakeup shapes.
- Test shim extended with BACKGROUND_BASH_ORPHAN and WAKEUP_ORPHAN scenarios.
- docs/cockpit.md updated to document off-protocol suppression rules and wakeup behavior.

Benefits
- Prevents false-positive prompt termination for legitimate background tasks and scheduled wakeups.
- Improves reliability and correctness of prompt lifecycle handling.
- Enables deterministic testing and simpler reasoning about watchdog behavior.
- Reduces risk of cross-prompt signal contamination and lost lifecycle updates.

[Review Change Stack](https://app.coderabbit.ai/change-stack/njbrake/agent-of-empires/pull/1406?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/cockpit/acp_client.rs [hunk 1]
```diff
@@ -336,56 +357,365 @@ fn classify_lifecycle_signal(
...
-    };
-    extract_tool_content_text(blocks).contains("Async agent launched successfully")
+) -> Option<OffProtocolWorkKind> {
+    let blocks = content.as_ref()?;
+    let text = extract_tool_content_text(blocks);
+    if text.contains("Async agent launched successfully") {
+        return Some(OffProtocolWorkKind::AsyncAgent);
+    }
+    if text.contains("Command running in background with ID: ") {
+        return Some(OffProtocolWorkKind::BackgroundCommand);
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/cockpit/acp_client.rs]: Fixed in 0e5de10c. `detect_off_protocol_work_completed` now iterates content blocks line-by-line and anchors with `trim_start().starts_with(...)`. New regression test `detect_off_protocol_work_completed_ignores_echoed_marker_mid_line` covers the echo case.

File: src/cockpit/acp_client.rs [hunk 2]
```diff
@@ -2458,7 +2812,58 @@ async fn run_connection_task<W, R>(
...
+                        mapped_events
+                            .iter()
+                            .filter_map(|e| match e {
+                                Event::WakeupScheduled { at, .. } => {
+                                    Some(LifecycleSignal::WakeupPending { at: *at })
+                                }
+                                _ => None,
+                            })
+                            .collect()
+                    };
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/cockpit/acp_client.rs]: Fixed in 0e5de10c. Replaced the `mapped_events.iter().filter_map` scan with a new `wakeup_lifecycle_signal_from_update` helper that gates on `ToolCallUpdate` with non-`Failed` status, title `ScheduleWakeup`, and parseable `raw_input.delaySeconds`. The initial `ToolCall` frame and any `Failed` completion no longer suppress the watchdog. Both `InProgress` and `Completed` are accepted because real `claude-agent-acp` populates `raw_input` on an interim update and strips it from the final `Completed` frame; requiring strictly `Completed` would lose the wakeup in production. New tests: `wakeup_lifecycle_signal_none_on_initial_tool_call`, `wakeup_lifecycle_signal_none_on_failed_completion`, `wakeup_lifecycle_signal_fires_on_in_progress_with_raw_input`, `wakeup_lifecycle_signal_from_completed_tool_call_update`. Test shim's WAKEUP_ORPHAN branch updated to emit an interim in_progress update with raw_input, mirroring real claude-agent-acp shape.

**Review by njbrake — APPROVED** [review]
_Note: this review was drafted by Claude via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

Approving. Clean fix for the documented production false-positive (#1240, #1360). The state-machine extraction makes the watchdog testable for the first time, and generalizing `ASYNC_AGENT_GRACE_FLOOR` to `OFF_PROTOCOL_WORK_GRACE_FLOOR` reuses existing machinery rather than adding a parallel path. The 30-minute finite ceiling preserves wedge recovery. New unit tests on the extracted struct plus the two integration scenarios cover the production shapes well; `EnvGuard` is a nice test-hygiene cleanup.

Optional follow-ups, none blocking:
- The comment in the notification handler (around lines 2878-2887) says the wakeup gate fires only on `Completed`; it now also accepts non-`Failed` `InProgress` updates. Trivial wording fix.
- `wakeup_event_from_raw` runs twice per `ScheduleWakeup` `ToolCallUpdate` (once from the lifecycle signal, once from `map_update_to_events`), so the `info!` log appears twice and `Utc::now()` is sampled twice. A pure `parse_wakeup_at(raw_input)` helper would dedup both, if you are already touching this code.
